libsolidity/parsing/Parser.cpp

```diff
@@ -1043,6 +1043,10 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 
 	yul::Parser asmParser(m_errorReporter, yul::EVMDialect::looseAssemblyForEVM());
 	shared_ptr<yul::Block> block = asmParser.parse(m_scanner, true);
+	if (block == nullptr)
+		// TODO: investigate if this can be done in a more cohesive manner.
```

Comment:
I think this is exactly the right way.

Reply:
The part confusing to me is this in `AsmParser`:
```
        catch (FatalError const&)
        {
                if (m_errorReporter.errors().empty())
                        throw; // Something is weird here, rather throw again.
        }
```

I think what this should be is an assertion instead and then it is consistent. `AsmParser` will not throw a `FatalError` exception ever, just a `nullptr` if unparseable - but may have assertions.

How do you feel changing and removing the comment in the solidity parser?

Reply:
we can change it to an assertion, yes.

Reply:
Which comment?